src/or/hs_client.c

```diff
@@ -1210,11 +1210,23 @@ hs_client_decode_descriptor(const char *desc_str,
   int ret;
   uint8_t subcredential[DIGEST256_LEN];
   ed25519_public_key_t blinded_pubkey;
+  hs_client_service_authorization_t *client_auth = NULL;
+  curve25519_secret_key_t *client_sk = NULL;
 
   tor_assert(desc_str);
   tor_assert(service_identity_pk);
   tor_assert(desc);
 
+  /* Check if we have a client authorization for this service in the map.
```

Comment:
Perhaps we should make this "Get client auth key given service identity" code into a function to make it more tidy? :)

Reply:
Sure